app/scripts/controllers/detect-tokens.js

```diff
@@ -116,7 +116,7 @@ export default class DetectTokensController {
     const tokensToDetect = [];
     for (const tokenAddress in tokenListUsed) {
       if (
-        !this.tokenAddresses.find(({ address }) =>
+        !this.tokenAddresses.find((address) =>
           isEqualCaseInsensitive(address, tokenAddress),
         ) &&
         !this.hiddenTokens.find((address) =>
```

Comment:
```suggestion
        !this.hiddenTokens.find(({ address }) =>
```

Reply:
`this.tokenAddresses` which is a list of token objects but `this.hiddenTokens` is a list of token addresses which is a list of strings.

Reply:
@NiranjanaBinoy I just want to confirm. Here you say "this.tokenAddresses which is a list of token objects", but above you changed the code to `!this.tokenAddresses.find((address) =>`. This seems to be a mismatch. Was there a mistake somewhere (either in the code or in your comment)?

Reply:
The mistake was in my comment; in the state, we are saving `tokens` as a list of objects, but in the constructor of `DetectTokensController,` we are filtering out the address to the `this.tokenAddresses` list. https://github.com/MetaMask/metamask-extension/blob/e73afb8451e7b055091f200ce00826ade6d0b924/app/scripts/controllers/detect-tokens.js#L74